src/org/exist/dom/QName.java

```diff
@@ -262,7 +262,7 @@ public static String extractLocalName(final String qname)
         } else if (p == qname.length()) {
             throw new IllegalArgumentException("Illegal QName: ends with a ':'");
         } else if (!isQName(qname)) {
-            throw new IllegalArgumentException("Illegal QName: not a valid local name.");
+            throw new IllegalArgumentException("Illegal QName: '" + qname + "' not a valid local name.");
```

Comment:
Yes - both showing the `qname` and surrounding it with quotes to make clear any extra whitespace.